src/FilenameParsing/HashFileIDHelper.php

```diff
@@ -79,7 +79,7 @@ public function cleanFilename($filename)
 
     public function parseFileID($fileID)
     {
-        $pattern = '#^(?<folder>([^/]+/)*)(?<hash>[a-zA-Z0-9]{10})/(?<basename>((?<!__)[^/.])+)(__(?<variant>[^.]+))?(?<extension>(\..+)*)$#';
+        $pattern = '#^(?<folder>([^/]+/)*)(?<hash>[a-f0-9]{10})/(?<basename>((?<!__)[^/.])+)(__(?<variant>[^.]+))?(?<extension>(\..+)*)$#';
```

Comment:
if we kept hash in the file ID by the variant, we could make the pattern even narrower, by comparing the hash through a back reference. E.g.:
```php
'#^(?<folder>([^/]+/)*)(?<hash>[a-f0-9]{10})/(?<basename>((?<!__)[^/.])+)(?:__\1)(__(?<variant>[^.]+))?(?<extension>(\..+)*)$#'
```